rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/terminal/LSPTerminalREPL.java

```diff
@@ -0,0 +1,129 @@
+    private static final InputStream stdin = System.in;
+    private static final OutputStream stderr = System.err;
+    private static final OutputStream stdout = System.out;
+    private static final boolean prettyPrompt = true;
+    private static final boolean allowColors = true;
+
+    public LSPTerminalREPL(Terminal terminal, IDEServices services) throws IOException, URISyntaxException {
+        super(makeInterpreter(terminal, services), null, stdin, stderr, stdout, true, terminal.isAnsiSupported(), getHistoryFile(), terminal, null);
+    }
+
+    private static ILanguageProtocol makeInterpreter(Terminal terminal, final IDEServices services) throws IOException, URISyntaxException {
+        
+        RascalInterpreterREPL repl =
+            new RascalInterpreterREPL(prettyPrompt, allowColors, getHistoryFile()) {
+                @Override
+                protected Evaluator constructEvaluator(InputStream input, OutputStream stdout, OutputStream stderr) {
+                    Evaluator eval =  ShellEvaluatorFactory.getDefaultEvaluator(input, stdout, stderr);
+
+                    URIResolverRegistry reg = URIResolverRegistry.getInstance();
+
+                    reg.registerLogical(new ILogicalSourceLocationResolver(){
+                        @Override
+                        public ISourceLocation resolve(ISourceLocation input) throws IOException {
+                            return services.resolveProjectLocation(input);
+                        }
+
+                        @Override
+                        public String scheme() {
+                            return "project";
+                        }
+
+                        @Override
+                        public String authority() {
+                            return null;
+                        }
+                    });
+                    
+                    return eval;
+                }
+
+                @Override
+                public void handleInput(String line, Map<String, InputStream> output, Map<String, String> metadata)
+                    throws InterruptedException {
+                    super.handleInput(line, output, metadata);
+
+                    for (String mimetype : output.keySet()) {
+                        if (!mimetype.contains("html") && !mimetype.startsWith("image/")) {
+                            continue;
+                        }
+
+                        services.browse(URIUtil.assumeCorrect(metadata.get("url")));
+                    }
+                }
+            };
+    
+        repl.setMeasureCommandTime(false);
+    
+        return repl;
+    }
+
+    private static File getHistoryFile() throws IOException {
+        File home = new File(System.getProperty("user.home"));
+        File rascal = new File(home, ".rascal");
+    
+        if (!rascal.exists()) {
+            rascal.mkdirs();
+        }
+    
+        File historyFile = new File(rascal, ".repl-history-rascal-terminal");
```

Comment:
I think we need to pick a different file name here, since this is the history inside the vs-code terminal.